server/chess.py

```diff
@@ -209,8 +195,7 @@ def get_knight_moves(
                 for file_direction in (-1, 1):
                     rank = knight.rank + rank_absolute * rank_direction
                     file = knight.file + file_absolute * file_direction
-                    victim = self.get_piece(rank, file)
-                    if (not victim) or (victim.side != knight.side):
+                    if validate_move(self, knight.rank, knight.file, rank, file):
```

Comment:
This method is slower - it re-fetches the piece from the database, and performs a whole lot of checks we have already done, for very little gain in code complexity.

Reply:
For starters it makes sure the knight doesn't jump off the board.

Reply:
Secondly it ensures the knight doesn't move to put his king in check

Reply:
The latter is already ensured on line 371, the former should be checked somewhere similar.